common/src/chain/config/mod.rs

```diff
@@ -567,30 +568,51 @@ fn create_testnet_genesis() -> Genesis {
     )
 }
 
-pub fn regtest_genesis_values() -> (
+pub fn regtest_genesis_values(
+    genesis_staking_settings: Option<String>,
+) -> (
     PoolId,
     Box<StakePoolData>,
     PrivateKey,
     PublicKey,
     VRFPrivateKey,
     VRFPublicKey,
 ) {
-    let genesis_pool_id =
-        decode_hex::<PoolId>("123c4c600097c513e088b9be62069f0c74c7671c523c8e3469a1c3f14b7ea2c4");
-
-    let genesis_stake_private_key = decode_hex::<PrivateKey>(
-        "008717e6946febd3a33ccdc3f3a27629ec80c33461c33a0fc56b4836fcedd26638",
-    );
-
-    let genesis_stake_public_key = decode_hex::<PublicKey>(
-        "0003c53526caf73cd990148e127cb57249a5e266d78df23968642c976a532197fdaa",
-    );
-
-    let genesis_vrf_private_key = decode_hex::<VRFPrivateKey>("003fcf7b813bec2a293f574b842988895278b396dd72471de2583b242097a59f06e9f3cd7b78d45750afd17292031373fddb5e7a8090db51221038f5e05f29998e");
-
-    let genesis_vrf_public_key = decode_hex::<VRFPublicKey>(
-        "00fa2f59dc7a7e176058e4f2d155cfa03ee007340e0285447892158823d332f744",
-    );
+    let genesis_staking_settings = genesis_staking_settings.unwrap_or_default();
+
+    let genesis_pool_id = Regex::new("genesis_pool_id:([A-Fa-f0-9]+)")
+        .expect("Regex is ok")
+        .captures(&genesis_staking_settings)
+        .and_then(|c| c.get(1))
+        .map(|c| c.as_str())
+        .or(Some(
+            "123c4c600097c513e088b9be62069f0c74c7671c523c8e3469a1c3f14b7ea2c4",
+        ))
+        .map(decode_hex::<PoolId>)
+        .expect("Parameter decoded {}");
```

Comment:
I find it a bit overkill to use a regex for this. How about just splitting the string on whitespace/comma and using `strip_prefix` to parse key-value entries, roughly like:
```rust
if let Some(pool_id_str) = chunk.strip_prefix("genesis_pool_id:") {
    pool_id = decode_hex(pool_id_str)?;
}
// Same for other settings
```

Reply:
I'm a regex fan, and it's only called once for the whole process, but no problem.